main.js

```diff
@@ -1,163 +1,312 @@
+}
+
+//2d. array of all items listed from lowest to highest price (in a function that is called
+//when low to high button is clicked. The buttons are to be found at the Bonus exercise)
+function lowestToHighestPrice() {
+    const lowestToHighestPrice = inventory.sort((a, b) => {
+        return a.price - b.price;
+    });
+    return document.getElementById("allProducts").innerHTML = allProducts(lowestToHighestPrice);
+}
+
+//3a. Target: Amount when all is sold-out ------------------------------
+function target() {
+    const targetPerItem = inventory.map((item) => {
+        return item.originalStock * item.price;
+    });
+
+    let targetCounter = 0;
+    for (let i = 0; i < targetPerItem.length; i++) {
+        targetCounter = targetCounter + targetPerItem[i];
+    }
+    return targetCounter.toString();
+}
+
+//returns the result to index.html
+document.getElementById("targetCounter").innerHTML = target();
+
+//3b. Counter: Current amount sold --------------------------------------
+function sold() {
+    const totalAmountSoldPerItem = inventory.map((item) => {
+        return (item.originalStock - item.sold) * item.price;
+    });
+
+    let counter = 0;
+    for (let i = 0; i < totalAmountSoldPerItem.length; i++) {
+        counter = counter + totalAmountSoldPerItem[i];
+    }
+    return counter.toString();
+}
+
+//returns the result to index.html
+document.getElementById("soldCounter").innerHTML = sold();
+
+//4. shows two type names ------------------------------------------------
+const typeNameOne = inventory[0].name;
+const typeNameTwo = inventory[1].name;
+
+//returns the result to index.html
+document.getElementById("inTheSpotLightsOne").innerHTML = typeNameOne;
+document.getElementById("inTheSpotLightsTwo").innerHTML = typeNameTwo;
+
+//5a function shows brand, type and name of one item as string ----------
+
+let randomItem = inventory[Math.floor(Math.random() * inventory.length)];
+
+function itemName(item) {
+    return item.brand + " " + item.type + " - " + item.name;
+}
+
+//returns result to index.html
+document.getElementById("itemName").innerHTML = itemName(randomItem);
+
+//5b function shows the price of one item as string with euro sign ------
+
+function itemPrice(item) {
+    return "€" + item.price + ",-";
+}
+
+//returns result to index.html
+document.getElementById("itemPrice").innerHTML = itemPrice(randomItem);
+
+//5c function calculates inches to cm and shows screen-sizes as string --
+function screenSizeInCm(item) {
+    const sizes = item.availableSizes.map((size) => {
+        return size + " inch (" + size * 2.54 + " cm)";
+    });
+    return sizes.join(" | ");
+}
+
+//returns result to index.html
+document.getElementById("screenSize").innerHTML = screenSizeInCm(randomItem);
+
+
+//5d using function a, b and c to present one item in box ---------------
+//already did this in 5a, b and c
+
+//5e tv generator function -----------------------------------------------
+function allProducts(list) {
```

Comment:
Dit is een hele mooie functie geworden. Ik heb zelf geen innerhtml gebruikt, maar dit lijkt veel mooier.